src/models/tests/auditModelGenerator.test.js

```diff
@@ -0,0 +1,363 @@
+import faker from 'faker';
+import { Model } from 'sequelize';
+import db, { User, ZALUser } from '..';
+import { auditLogger } from '../../logger';
+import audit from '../auditModelGenerator';
+
+describe('Audit System', () => {
+  let t;
+  let transactionVariables;
+
+  beforeEach(async () => {
+    t = await db.sequelize.transaction();
+    transactionVariables = {
+      loggedUser: `${faker.datatype.number()}`,
+      transactionId: faker.datatype.uuid(),
+      sessionSig: faker.datatype.string(32).replace(/[^a-zA-Z0-9!@#$%^&*()_+,.<>?;:]/g, ''),
+      auditDescriptor: 'Audit System Test',
+    };
+
+    const query = `SELECT
+      set_config('audit.loggedUser', '${transactionVariables.loggedUser}', TRUE) as "loggedUser",
+      set_config('audit.transactionId', '${transactionVariables.transactionId}', TRUE) as "transactionId",
+      set_config('audit.sessionSig', '${transactionVariables.sessionSig}', TRUE) as "sessionSig",
+      set_config('audit.auditDescriptor', '${transactionVariables.auditDescriptor}', TRUE) as "auditDescriptor";`;
+
+    await db.sequelize.queryInterface.sequelize.query(
+      query,
+      { transaction: t },
+    );
+  });
+
+  afterEach(async () => {
+    if (t) {
+      // await t.commit();
+      await t.rollback();
+    }
+  });
+
+  afterAll(async () => {
+    await db.sequelize.close();
+  });
+
+  describe('General Audit System', () => {
+    it('Transaction variables', async () => {
+      const query = `select
+      current_setting('audit.loggedUser', true) as "loggedUser",
+      current_setting('audit.transactionId', true) as "transactionId",
+      current_setting('audit.sessionSig', true) as "sessionSig",
+      current_setting('audit.auditDescriptor', true) as "auditDescriptor";`;
+
+      const values = await db.sequelize.queryInterface.sequelize.query(
+        query,
+        {
+          type: db.sequelize.QueryTypes.SELECT,
+          transaction: t,
+        },
+      );
+
+      expect(values[0]).toEqual(transactionVariables);
+    });
+
+    describe('Automatic Audit New Tables', () => {
+      it('Audit Triggers Automatic added to New Tables', async () => {
+        class Test extends Model {}
+
+        Test.init(
+          {
+            id: {
+              allowNull: false,
+              primaryKey: true,
+              type: db.Sequelize.BIGINT,
+              autoIncrement: true,
+            },
+            value: {
+              allowNull: true,
+              default: null,
+              type: db.Sequelize.STRING,
+            },
+          },
+          {
+            sequelize: db.sequelize,
+            createdAt: false,
+            updatedAt: false,
+            transaction: t,
+          },
+        );
+
+        await Test.sync({ force: true, alter: true, transaction: t })
+          .catch((err) => auditLogger.error(err));
```

Comment:
If we use `await` we need to wrap the call in a `try catch` instead of adding a `.catch` at the end of the promise chain. Adding the `.catch` at the end mixes two ways of dealing with promises, chaining vs await/async.